docs/traits.md

```diff
@@ -0,0 +1,153 @@
+template <>
+struct Access<PointCloud, PrimitivesTag>
+{
+    inline std::size_t size(PointCloud const& cloud)
+    {
+        return cloud.N;
+    }
+      KOKKOS_INLINE_FUNCTION Point get(PointCloud const& cloud, std::size_t i)
+    {
+        return {{cloud.d_x[i], cloud.d_y[i], cloud.d_z[i]}};
+    }
+    using memory_space = Kokkos::CudaSpace;
+};
+
+template <>
+struct Access<PointCloud, PredicatesTag>
+{
+    inline std::size_t size(PointCloud const& cloud)
+    {
+        return cloud.N;
+    }
+      KOKKOS_INLINE_FUNCTION auto get(PointCloud const& cloud, std::size_t i)
+    {
+        float const radius = 10.0;
+        Point const point = {{cloud.d_x[i], cloud.d_y[i], cloud.d_z[i]}};
+        return intersects({point, radius});
+    }
+    using memory_space = Kokkos::CudaSpace;
+};
+} // namespace Traits
+} // namespace ArborX
+
+std::ostream& operator<<(std::ostream& os, ArborX::Point const& p)
+{
+    os << '(' << p[0] << ',' << p[1] << ',' << p[2] << ')';
+}
+
+int main(int argc, char *argv[])
+{
+    Kokkos::initialize(argc, argv);
+    
+    constexpr std::size_t N = 1000;
+    std::array<float, N> a;
+
+    float *d_a;
+    cudaMalloc(&d_a, sizeof(float) * N);
+
+    std::iota(std::begin(a), std::end(a), 1.0);
+
+    cudaMemcpy(d_a, a.data(), sizeof(float) * N, cudaMemcpyHostToDevice);
+
+    ArborX::BVH<Kokkos::Cuda::device_type> bvh{PointCloud{d_a, d_a, d_a, N}};
+
+    auto const box = bvh.bounds();
+    std::cout << box.min_corner() << " - " << box.max_corner() << '\n';
+
+    Kokkos::finalize()
+
+    return 0;
+}
+```
+Output
+```
+(1,1,1) - (1000,1000,1000)
+```
+
+# `ArborX::Traits::Access<T,Tag>::size`
+```C++
+size_type size(T const& data);
+```
+## Parameters
+`data`
+: data argument passed to the `ArborX::BVH<DeviceType>::BVH` constructor or `ArborX::BVH<DeviceType>::query()`
+## Returns
+The number of primitives if the second template argument is `PrimitivesTag` or the number of queries to perform if it is `PredicatesTag`.
+
+# `ArborX::Traits::Access<T,Tag>::get`
+```C++
+KOKKOS_FUNCTION context_dependent get(T const& data, size_type pos);
+```
+If `std::is_same<Tag,PrimitimesTag>::value` is `true`, the return value type must either decay to `ArborX::Point` or to `ArborX::Box`.
+
+If `std::is_same<Tag,PredicatesTag>::value` is `true`, the return value type must decay to a valid ArborX predicate, e.g. a predicate generated by `ArborX::nearest()` or `ArborX::intersects()`.
+
+`get()` needs to be marked with the `KOKKOS_FUNCTION` or `KOKKOS_INLINE_FUNCTION` macro because it is called from the device.
+## Parameters
+`data`
+: data argument passed to the `ArborX::BVH<DeviceType>::BVH` constructor or `ArborX::BVH<DeviceType>::query()`  
+`pos`
+: position of the primitive from which to return the centroid or the smallest bounding volume
+## Returns
+
+## Notes
+Will be called exactly once for every `pos` in the half-open range `[0, N)` where `N = ArborX::Traits::Access<Primitives,PrimitivesTag>::size(primitives)`.
```

Comment:
Why put this note in? This is an additional restriction that users may start relying on. Does that easy the use for users somehow? I would remove this note to leave room for future if we ever want to change that.